waddrmgr/manager_test.go

```diff
@@ -1019,38 +1019,64 @@ func testImportScript(tc *testContext) bool {
 
 // testMarkUsed ensures used addresses are flagged as such.
 func testMarkUsed(tc *testContext) bool {
-	expectedAddr1 := expectedAddr{
-		addressHash: hexToBytes("2ef94abb9ee8f785d087c3ec8d6ee467e92d0d0a"),
-		used:        true,
+	tests := []struct {
+		name       string
+		addrType   btcutil.Address
+		in         string
+		blockstamp waddrmgr.BlockStamp
+		expected   expectedAddr
+	}{
+		{
+			name:     "managed address",
+			addrType: &btcutil.AddressPubKeyHash{},
+			in:       "1HZwkjkeaoZfTSaJxDw6aKkxp45agDiEzN",
+			expected: expectedAddr{
+				addressHash: hexToBytes("2ef94abb9ee8f785d087c3ec8d6ee467e92d0d0a"),
+				used:        true,
+			},
+		},
+		{
+			name:     "script address",
+			addrType: &btcutil.AddressPubKeyHash{},
+			in:       "3MbyWAu9UaoBewR3cArF1nwf4aQgVwzrA5",
+			expected: expectedAddr{
+				addressHash: hexToBytes("da6e6a632d96dc5530d7b3c9f3017725d023093e"),
+				used:        true,
+			},
+		},
 	}
+
 	prefix := "MarkUsed"
 	chainParams := tc.manager.ChainParams()
-	addrHash := expectedAddr1.addressHash
-	addr, err := btcutil.NewAddressPubKeyHash(addrHash, chainParams)
+	for i, test := range tests {
+		addrHash := test.expected.addressHash
+		addr, err := btcutil.NewAddressPubKeyHash(addrHash, chainParams)
 
-	if tc.create {
-		// Test that initially the address is not flagged as used
+		if tc.create {
+			// Test that initially the address is not flagged as used
+			maddr, err := tc.manager.Address(addr)
+			if err != nil {
+				tc.t.Errorf("%s #%d: unexpected error: %v", prefix, i, err)
+			}
+			if maddr.Used() != false {
+				tc.t.Errorf("%s #%d: unexpected used flag -- got "+
+					"%v, want %v", prefix, i, maddr.Used(), test.expected.used)
+			}
+		}
+		err = tc.manager.MarkUsed(addrHash)
+		if err != nil {
+			tc.t.Errorf("%s #%d: unexpected error: %v", prefix, i, err)
+		}
 		maddr, err := tc.manager.Address(addr)
 		if err != nil {
-			tc.t.Errorf("%s: unexpected error: %v", prefix, err)
+			tc.t.Errorf("%s #%d: unexpected error: %v", prefix, i, err)
 		}
-		if maddr.Used() != false {
-			tc.t.Errorf("%v: unexpected used flag -- got "+
-				"%v, want %v", prefix, maddr.Used(), expectedAddr1.used)
+		if maddr.Used() != test.expected.used {
+			tc.t.Errorf("%s #%d: unexpected used flag -- got "+
+				"%v, want %v", prefix, i, maddr.Used(), test.expected.used)
 		}
```

Comment:
I would consider closing the db here, and reopening, and check that it is still marked as used.  Perhaps do the same shutdown dance before testing for initially unsued as well.